client/me/two-step/index.jsx

```diff
@@ -46,60 +48,20 @@ class TwoStep extends Component {
 		doingSetup: false,
 	};
 
-	componentDidMount() {
-		debug( this.constructor.displayName + ' React component is mounted.' );
-		this.props.userSettings.on( 'change', this.onUserSettingsChange );
-		this.props.userSettings.fetchSettings();
+	isInitialized() {
+		return !! this.props.userSettings;
 	}
 
-	componentWillUnmount() {
-		debug( this.constructor.displayName + ' React component is unmounting.' );
-		this.props.userSettings.off( 'change', this.onUserSettingsChange );
+	isDoingSetup() {
+		return ! this.props.isTwoStepEnabled;
```

Comment:
`isTwoStepEnabled` will be `null` if data isn't fetched yet. So in that case, `isDoingSetup` will return false during data fetch. Is this expected?

Reply:
`! null` will resolve to `true`, no? That's still different from what we had before (it used to be initialized with `false`) but that shouldn't be a problem afaics 🤔

Reply:
Yep, I meant that `isDoingSetup` will return `true` because `this.props.isTwoStepEnabled` would be evaluated to false. 😉  

But still, I find it odd that we make the assumption that setup is in progress before data has actually been fetched - what are your thoughts about that?

Reply:
I agree, although I think that's mostly a naming issue. I pushed some changes where I remove `isInitialized` and `isDoingSetup` by introducing a `fetching` reducer to the user settings state (44b4594 and 2454ea1). Let me know what you think :)